apps/mobile/src/features/send/send-sheets/send-form-btc-sheet.tsx

```diff
@@ -1,38 +1,22 @@
-import { FormProvider, useForm } from 'react-hook-form';
-
 import { FullHeightSheetHeader } from '@/components/full-height-sheet/full-height-sheet-header';
 import { FullHeightSheetLayout } from '@/components/full-height-sheet/full-height-sheet.layout';
 import { NetworkBadge } from '@/features/settings/network-badge';
-import { useBitcoinAccountTotalBitcoinBalance } from '@/queries/balance/bitcoin-balance.query';
-import { zodResolver } from '@hookform/resolvers/zod';
 import { t } from '@lingui/macro';
 import { useLingui } from '@lingui/react';
 
 import { BtcAvatarIcon } from '@leather.io/ui/native';
 
-import {
-  SendFormBtcSchema,
-  defaultSendFormBtcValues,
-  sendFormBtcSchema,
-} from '../send-form/schemas/send-form-btc.schema';
+import { CreateCurrentSendRoute, useSendSheetRoute } from '../send-form.utils';
+import { useSendFormBtc } from '../send-form/hooks/use-send-form-btc';
+import { SendFormBtcProvider } from '../send-form/providers/send-form-btc-provider';
 import { SendForm } from '../send-form/send-form';
-import { CreateCurrentSendRoute, useSendSheetNavigation, useSendSheetRoute } from '../utils';
 
 type CurrentRoute = CreateCurrentSendRoute<'send-form-btc'>;
 export function SendFormBtcSheet() {
   const { i18n } = useLingui();
   const route = useSendSheetRoute<CurrentRoute>();
-  const navigation = useSendSheetNavigation<CurrentRoute>();
-
-  const formMethods = useForm<SendFormBtcSchema>({
-    defaultValues: defaultSendFormBtcValues,
-    resolver: zodResolver(sendFormBtcSchema),
-  });
 
-  const { availableBalance, fiatBalance } = useBitcoinAccountTotalBitcoinBalance({
-    accountIndex: route.params.account.accountIndex,
-    fingerprint: route.params.account.fingerprint,
-  });
+  const { onGoBack } = useSendFormBtc();
```

Comment:
Alternative to the loader, wonder if async dependencies & loading states of SendFormBtcProvider could be handled here?

Reply:
Went with the loader pattern, but happy to revisit.